inference-engine/src/mkldnn_plugin/nodes/mkldnn_topk_node.cpp

```diff
@@ -40,190 +1230,466 @@ bool MKLDNNTopKNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& o
+}
 
-    SizeVector in_dims = getParentEdgeAt(TOPK_DATA)->getDims().ToSizeVector();
+inline void MKLDNNTopKNode::topk_kernel_process(const uint8_t *in_p, uint8_t *out_p, uint8_t *out_idx_p,
+                                                uint8_t *process_p, uint8_t *process_idx_p, size_t work_amount) {
+    auto arg = jit_topk_call_args();
+    arg.src = static_cast<const void *>(in_p);
+    arg.process = static_cast<void *>(process_p);
+    arg.process_index = static_cast<void *>(process_idx_p);
+    arg.dst = static_cast<void *>(out_p);
+    arg.index = static_cast<void *>(out_idx_p);
+    arg.work_amount = work_amount;
+    (*topk_kernel)(&arg);
+}
 
-    if (src_k == 1) {
+void MKLDNNTopKNode::topk_ref(const float *in_ptr, float *out_ptr, int32_t *dst_idx) {
+    if (top_k == 1) {
         if (is_last_dim) {
             if (mode_max)
-                top1<std::greater>(src, dst_data, dst_idx, in_dims);
+                top1(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x > y; });
             else
-                top1<std::less>(src, dst_data, dst_idx, in_dims);
+                top1(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x < y; });
         } else {
             if (mode_max)
-                top1_axis<cmpgt_ps, std::greater>(src, dst_data, dst_idx, in_dims);
+                top1_axis(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x > y; });
             else
-                top1_axis<cmplt_ps, std::less>(src, dst_data, dst_idx, in_dims);
+                top1_axis(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x < y; });
         }
     } else {
         if (is_last_dim) {
             if (mode_max)
-                topk<std::greater>(src, dst_data, dst_idx, in_dims);
+                topk(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x > y; });
             else
-                topk<std::less>(src, dst_data, dst_idx, in_dims);
+                topk(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x < y; });
         } else {
             if (mode_max)
-                topk_axis<cmpgt_ps, std::greater>(src, dst_data, dst_idx, in_dims);
+                topk_axis(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x > y; });
             else
-                topk_axis<cmplt_ps, std::less>(src, dst_data, dst_idx, in_dims);
+                topk_axis(in_ptr, out_ptr, dst_idx, src_dims, [](float x, float y)->float { return x < y; });
         }
     }
 }
 
-bool MKLDNNTopKNode::created() const {
-    return getType() == TopK;
-}
-
-template <class Compare1, template <typename> class Compare2>
-void MKLDNNTopKNode::top1_axis(const float* src_data, float* dst_data, int* dst_idx, SizeVector in_dims) {
+void MKLDNNTopKNode::top1_axis(const float* src_data, float* dst_data, int32_t* dst_idx, SizeVector in_dims, std::function<float(float, float)> compare) {
```

Comment:
SizeVector in_dims can be replaced with const SizeVector&

Reply:
Fixed. Thanks Egor